routes/containers/index.js

```diff
@@ -670,6 +670,201 @@ router.post(
     }
 );
 
+/**
+ * @apiName Containers renew container
+ * @apiGroup Containers
+ *
+ * @api {post} /containers/renew/:id Renew specific container
+ * @apiPermission bot
+ * @apiPermission clerk
+ * @apiPermission admin
+ * 
+ * @apiUse JWT_orderTime
+ * @apiSuccessExample {json} Success-Response:
+        HTTP/1.1 200 
+        {
+            type: "ReturnMessage",
+            message: "Return Succeeded",
+            oriUser: "09xxxxxxxx",
+            containerList: [ 
+                {
+                    typeName: String,
+                    typeCode: Number,
+                    id: Number
+                },
+                ...
+            ]
+        }
+ * @apiUse RenewError
+ * @apiUse ChangeStateError
+ */
+router.post('/renew/:id',
+    regAsBot,
+    regAsStore,
+    regAsAdmin,
+    validateRequest,
+    function(req, res, next) {
+        let dbStore = req._user;
+        if (!res._payload.orderTime)
+            return res.status(403).json({
+                code: 'F006',
+                type: 'renewContainerMessage',
+                message: 'Missing Order Time'
+            });
+        let container = req.params.id;
+        const storeID = req.body.storeId;
+        let activity = req.params.activity || null;
+        if (container === 'list') container = req.body.containers;
+        else container = [container];
+        changeContainersState(
```

Comment:
從這邊開始的 callback hell 有點太恐怖了，建議用下面的方式包裝一下會比較好閱讀
```suggestion
        new Promise((resolve, reject) => {
            /* 
            .
            .
            .
            some codes ....
            */
            someFunction(param1, param2, () => {
                resolve(someValue)
            })
        })
```